soroban-env-host/src/native_contract/base_types.rs

```diff
@@ -246,6 +248,13 @@ impl<const N: u32> BytesN<N> {
     }
 }
 
+impl<const N: u32> MeteredClone for BytesN<N> {
+    fn metered_clone(&self, budget: &Budget) -> Result<Self, HostError> {
+        budget.charge(CostType::BytesClone, N.into())?;
+        Ok(self.clone())
```

Comment:
This `self.clone()` is not actually cloning anything host side, so I don't think we should be charging for it. If you see here this is the type, and it is just an `Object`, so when we clone we're just cloning a u64 value: https://github.com/stellar/rs-soroban-env/blob/0f2119839fb339e17bb592ea695317b3bf590127/soroban-env-host/src/native_contract/base_types.rs#L160-L161

Reply:
I think this comment would apply to a lot of the types and clones in the native contract.

Reply:
Thanks @leighmcculloch, good catch! I updated it. 
In which case, there isn't really much changes left in this PR (which is great!) since most operations are trivial/already covered by the host.